### PR TITLE
Document <files> and <execfiles>.

### PR DESCRIPTION
**Pull request by ProgVal**
Closes GH-65.

As you know I'm working on a PR to show fields added by modules, and these two are a big use-case.

### CODE REVIEW

File: docs/3/configuration/_execfiles.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  Note that commands are only run on rehash, so changes are not automatically detected.
+attributes:
+- name: motd
+  type: Text
+  required: false
+  default: None
+  description: |-
+    A file used as the Message Of The Day, ie. shown to users when they connect or use the `/MOTD` command.
+details: |-
+  The working directory of the commands is InspIRCd's configuration directory. This is `/etc/inspircd` if installed with your package manager, and the main `conf/` directory otherwise. It is **NOT** the directory of the current config file.
```
Comment (SadieCat) [inline docs/3/configuration/_execfiles.yml:14]: `<execfiles>` is not relative to anything as far as I know.

File: docs/3/configuration/_execfiles.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+name: <execfiles>
+description: |-
+  The `<execfiles>` runs a shell command, and reads its output. The purpose of the output depends on what field commands are defined in.
```
Comment (SadieCat) [inline docs/3/configuration/_execfiles.yml:3]:
```suggestion
  The `<execfiles>` tag runs a shell command and reads its output. This tag can only be defined once.
  
  The purpose of the output depends on what field commands are defined in.
  
```

... probably?

File: docs/3/configuration/_files.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+name: <files>
+description: |-
+  The `<files>` tag reads one or more files, usually long text files. The purpose of these files depends on what field they are defined in.
```
Comment (SadieCat) [inline docs/3/configuration/_files.yml:3]:
```suggestion
  The `<files>` tag reads one or more files, usually long text files. This tag can only be defined once.
  
  The purpose of these files depends on what field they are defined in.

```

File: docs/3/configuration/_files.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  Note that files are only read on rehash, so changes are not automatically detected.
+attributes:
+- name: motd
+  type: Text
+  required: false
+  default: None
+  description: |-
+    A file used as the Message Of The Day, ie. shown to users when they connect or use the `/MOTD` command.
+details: |-
+  Relative paths are relative to InspIRCd's configuration directory. This is `/etc/inspircd` if installed with your package manager, and the main `conf/` directory otherwise. They are **NOT** relative to the current config file.
```
Comment (SadieCat) [inline docs/3/configuration/_files.yml:17]:
```suggestion
  All paths are relative to InspIRCd's configuration directory. This is `/etc/inspircd` if using our binary packages and defaults to `./run/conf` if you have built from source. They are **NOT** relative to the current config file.
```

File: docs/3/configuration/_execfiles.yml (hunk 1 above)
Comment (ProgVal) [reply docs/3/configuration/_execfiles.yml:17]: It has to have a CWD

Comment (SadieCat) [reply docs/3/configuration/_execfiles.yml:17]: The IRCd does not enforce any particular working directory is what I mean.

Comment (ProgVal) [reply docs/3/configuration/_execfiles.yml:17]: Indeed, I missed that it explicitly prepends the prefix to <files> so I assumed it set the CWD. My bad

File: docs/3/configuration/_execfiles.yml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+name: <execfiles>
+description: |-
+  The `<execfiles>` tag runs a shell command and reads its output. This tag can only be defined once.
+
+  The purpose of the output depends on what field commands are defined in.
```
Comment (SadieCat) [inline docs/3/configuration/_execfiles.yml:5]:
```suggestion
  The `<execfiles>` tag runs a shell command and reads its output. This tag can only be defined once. The purpose of the output depends on what field commands are defined in.
```

File: docs/3/configuration/_execfiles.yml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  Note that commands are only run on rehash, so changes are not automatically detected.
+attributes:
+- name: motd
+  type: Text
+  required: false
+  default: None
+  description: |-
+    A file used as the Message Of The Day, ie. shown to users when they connect or use the `/MOTD` command.
+details: |-
+  The working directory of the directory use to run InspIRCd.
```
Comment (SadieCat) [inline docs/3/configuration/_execfiles.yml:16]: What's this for?

File: docs/3/configuration/_execfiles.yml [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+example: |-
+  Downloads `https://www.example.com/motd.txt` and uses it as Message Of The Day:
```
Comment (SadieCat) [inline docs/3/configuration/_execfiles.yml:18]:
```suggestion
  Downloads `https://www.example.com/motd.txt` and uses it as the message of the day:
```

(We don't capitalise this.)

File: docs/3/configuration/_files.yml [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+  The purpose of these files depends on what field they are defined in.
+
+  Note that files are only read on rehash, so changes are not automatically detected.
+attributes:
+- name: motd
+  type: Text
+  required: false
+  default: None
+  description: |-
+    A file used as the Message Of The Day, ie. shown to users when they connect or use the `/MOTD` command.
```
Comment (SadieCat) [inline docs/3/configuration/_files.yml:13]:
```suggestion
    A file used as the message of the day, i.e. shown to users when they connect or use the `/MOTD` command.
```

File: docs/3/configuration/_execfiles.yml (hunk 6 above)
Comment (ProgVal) [reply docs/3/configuration/_execfiles.yml:16]: I meant "The working directory **is** the directory **used** to run InspIRCd."

File: docs/3/configuration/_files.yml [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+- name: motd
+  type: Text
+  required: false
+  default: None
+  description: |-
+    A file used as the message of the day, i.e. shown to users when they connect or use the `/MOTD` command.
+details: |-
+  All paths are relative to InspIRCd's configuration directory. This is `/etc/inspircd` if using our binary packages and defaults to `./run/conf` if you have built from source. They are **NOT** relative to the current config file.
+example: |-
+  Use the file `examples/motd.txt.example` as message of the day:
```
Comment (SadieCat) [inline docs/3/configuration/_files.yml:16]:
```suggestion
  Use the file `examples/motd.txt.example` as the message of the day:
```

**Pull request merged**